### PR TITLE
PR: Update donation/funding status message in Readme

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] ~~Wrote at least one-line docstrings for any new functions~~
* [x] ~~Added at least one unit test covering the changes, if at all possible~~
* [x] Described your changes and the motivation for them below
* [x] ~~Noted what issue(s) this pull request resolves, creating one if needed~~
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

@ccordoba12 

Modifies the funding message to better reflect the current situation, shortens it to be more concise to take up less space at the top of the readme, and makes it more robust to status changes up to the planned Spyder 4 final release. Can/will also be applied to ``spyder-docs``, ``QtPy`` and ``spyder-unittest`` (as actively supported core or plugin projects that have the old message) once approved here. 

image

Meanwhile, the version to replace the current one on the Spyder plugins (``spyder-xxx`` other than ``spyder-unittest``) would have a heavily modified first paragraph making clear the current development pause and the plan to restart it after the Spyder 4 release (absent specific funding), while keeping essentially the same other two paragraphs. I further propose dropping the message entirely from our affiliated projects (QtAwesome, QtSSAS, PyWinPTY, etc.) that aren't core parts of Spyder development, unless specifically warranted since they would not be directly affected/funded by the donations.

Finally, I added a new [Current Funding and Development Status page](https://github.com/spyder-ide/spyder/wiki/Current-Funding-and-Development-Status) on the wiki, and linked to it instead of the old one, to accomplish similar aims (to update the current situation, make it more robust to future changes, and make the tone more positive along the same lines as #6923 ).

### Issue(s) Resolved

Minor meta-task with no user-visible changes; none needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,30 +23,25 @@
...
-toward an eventual Spyder 4 feature release sometime in the future.
-
-However, with your contribution of effort and funding, we will be able to
-both continue to maintain Spyder 3 at a faster pace, and fund development of
-new features for Spyder 4 (such as a major overhaul to improve
-code completion, and a much-improved new debugger, both of which you've been
-requesting) at a greatly accelerated rate.
-
-There are many ways to [help with development](
+## Help support the community-developed scientific IDE!
```
Comment (ccordoba12) [inline README.md]: Let's change this to

> Help support Spyder, the community-...

to make things easier to understand

**Comment by ccordoba12** [comment]
@CAM-Gerlach, could you open PRs in all repos that have the funding status message and remove it? I don't want to be updating that message in a multitude of repos, just here.

Thanks!

**Comment by ccordoba12** [comment]
> Meanwhile, the version to replace the current one on the Spyder plugins (spyder-xxx other than spyder-unittest) would have a heavily modified first paragraph making clear the current development pause and the plan to restart it after the Spyder 4 release (absent specific funding), while keeping essentially the same other two paragraphs

Sorry, I didn't see this before my last message. I agree with it.

> I further propose dropping the message entirely from our affiliated projects (QtAwesome, QtSSAS, PyWinPTY, etc.) that aren't core parts of Spyder development, unless specifically warranted since they would not be directly affected/funded by the donations.

I agree with this too.

> Finally, I added a new Current Funding and Development Status page on the wiki

Great! Thanks a lot for that.

File: README.md (hunk 1 above)
Comment (CAM-Gerlach) [reply README.md]: Sure; done.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Tweaked the section title per your request, and will go ahead with the above plan.

The idea is to make it more general, so that until there is a major change in funding status of an individual project that would warrant an announcement anyway (like Spyder 4's release and restarting development of certain plugins) it shouldn't need to be updated, and they all point to the  centralized wiki page where more detailed updates can be posted as needed, and that in turn to the blog to get the latest status.

**Comment by CAM-Gerlach** [comment]
I'll submit a PR to ``spyder-docs`` since the funding may end up covering that and I can maintain it, and I won't add it to the website and remove it from QtPy and everywhere but our main plugins with paused development.

Also, I will submit a PR to ``spyder-terminal`` first to make sure the plugin specific text is good with you, and then I'll do the other PRs to add it to the existing paused plugins and remove it everywhere else.

**Pull request merged**